crates/nu-parser/src/parser.rs

```diff
@@ -2717,34 +2717,23 @@ pub fn parse_shape_name(
         b"bool" => SyntaxShape::Boolean,
         b"cell-path" => SyntaxShape::CellPath,
         b"closure" => SyntaxShape::Closure(None), //FIXME: Blocks should have known output types
-        b"cond" => SyntaxShape::RowCondition,
-        // b"custom" => SyntaxShape::Custom(Box::new(SyntaxShape::Any), SyntaxShape::Int),
         b"datetime" => SyntaxShape::DateTime,
         b"directory" => SyntaxShape::Directory,
         b"duration" => SyntaxShape::Duration,
         b"error" => SyntaxShape::Error,
-        b"expr" => SyntaxShape::Expression,
         b"float" => SyntaxShape::Float,
         b"filesize" => SyntaxShape::Filesize,
-        b"full-cell-path" => SyntaxShape::FullCellPath,
+        b"full-cell-path" => SyntaxShape::FullCellPath, // TODO: Should users distinguish cellpath?
         b"glob" => SyntaxShape::GlobPattern,
         b"int" => SyntaxShape::Int,
-        b"import-pattern" => SyntaxShape::ImportPattern,
-        b"keyword" => SyntaxShape::Keyword(vec![], Box::new(SyntaxShape::Any)),
         _ if bytes.starts_with(b"list") => parse_list_shape(working_set, bytes, span),
-        b"math" => SyntaxShape::MathExpression,
```

Comment:
a question that applies to all the others being removed:

if we do not use such a shape, could we remove it from the code base entirely? e.g. could we remove `SyntaxShape::MathExpression`? :smirk:

Reply:
`SyntaxShape` does more than `Type` and `MathExpression` tells the parser to parse a math expression. While useless for custom commands it is used by the `let`/`mut`/`const` "commands" to properly parse the assignment.

Reply:
ooh ok gotcha, so let's keep them for internals :relieved: